From 064fef404f69582c84c1f32123186c81fb2d8ddc Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Thu, 11 May 2017 15:49:12 +0800
Subject: [PATCH 1000/1345] fix: a3700: disable u3d driver in SGMII1 board
 configuration

commit  19f1d07721fdd4656c7f3cd2db0f3d70ed5fdd31 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

USB3 and SGMII-1 share the serdes lane, so they cannot work
concurrently. In the dts file of SGMII-1, usb3 device driver
should be disabled together with usb3 host driver. Otherwise,
SGMII-1 cannot work properly in some cases SGMII phy requires
reinitialization, e.g., resume from system suspend.

Change-Id: I3a6e7e844f6ca913fbdea14523a7c47402e626a0
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39322
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39372
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db-B.dts |    3 +++
 arch/arm64/boot/dts/marvell/armada-3720-db-C.dts |    3 +++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts b/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
index 75b5013..c8db25c 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db-B.dts
@@ -61,6 +61,9 @@
 				bm,pool-long = <2>;
 				bm,pool-short = <3>;
 			};
+			u3d@50000 {
+				status = "disabled";
+			};
 			usb3@58000 {
 				status = "disabled";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db-C.dts b/arch/arm64/boot/dts/marvell/armada-3720-db-C.dts
index 1adfb2a..d7008b7 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db-C.dts
@@ -64,6 +64,9 @@
 					full-duplex;
 				};
 			};
+			u3d@50000 {
+				status = "disabled";
+			};
 			usb3@58000 {
 				status = "disabled";
 			};
-- 
1.7.9.5

